spec/syncSpec.js

```diff
@@ -105,53 +113,56 @@ describe('Sync', function() {
 
   it('should delete', function() {
     return manage(datasetId)
-    .then(doCreate(datasetId, testData))
-    .then(doDelete())
-    .then(doRead(datasetId))
-    .catch(function(err) {
-      expect(err).toEqual('unknown_uid');
-    });
+      .then(doCreate(datasetId, testData))
+      .then(function withResult(res) {
+        const uid = res.uid;
+        return doDelete(datasetId, uid)
+        .then(doRead(datasetId, uid).catch(function(err) {
+          expect(err).toEqual('unknown_uid');
+          console.log(err);
+        }));
+      });
   });
 
   it('should cause a collision', function() {
     const collisionData = { test: 'cause a collision' };
-    // The UID of the record which should have a collision.
-    var recordId;
 
     return manage(datasetId)
     .then(doCreate(datasetId, testData))
-    .then(waitForSyncEvent('remote_update_applied'))
-    .then(function verifyUpdateApplied(event) {
-      // We need to store this for updating in MongoDB in the next step.
-      recordId = event.message.uid;
-      expect(recordId).not.toBeNull();
-      return recordId;
-    })
-    .then(updateRecord(datasetId, collisionData))
-    .then(doUpdate())
-    .then(waitForSyncEvent('collision_detected'))
-    .then(function verifyCorrectCollision(event) {
-      // Assert that the collision is the one we caused.
-      expect(event.message.uid).toEqual(recordId);
-    })
-    .then(listCollisions)
-    .then(function verifyCollisionInList(collisions) {
-      // Find the collision we invoked earlier.
-      const invokedCollision = searchObject(collisions, function(collision) {
-        return collision.uid === recordId;
+    .then(function withResult() {
+      return waitForSyncEvent('remote_update_applied')()
```

Comment:
I might be reading this incorrectly, but we don't seem to use the result provided by `doCreate` here, so could this be changed to the following (removing the first indent)?

```javascript
return manage(datasetId)
.then(doCreate(datasetId, testData))
.then(waitForSyncEvent('remote_update_applied'))
.then(function verifyUpdateApplied() {
  // We need to store this for updating in MongoDB in the next step.
  expect(event.message.uid).not.toBeNull();
  ...
})
```